tests/README.md

```diff
@@ -1,24 +1,130 @@
+- Raise a cluster based on the configuration.
+  At this stage, it will not contain kubevirt.
+  ```
+  make cluster-up
+  ```
+- Build KubeVirt from sources.
+  ```
+  make
+  ```
+- Deploy KubeVirt onto the cluster.
+  ```
+  make cluster-sync
+  ```
+- Run the tests.
+  ```
+  make functest
+  ```
+- Take down and delete the cluster.
+  ```
+  make cluster-down
+  ```
+
+On a regular workflow of development or debugging, rebuilding KubeVirt is
+needed only when the production code is changed.
+In case only test code changes, `make functest` will rebuild the test code
+implicitly.
+
+## Existing (external) Kubevirt Cluster
+
+The tests may be executed against an already deployed (and active) Kubevirt
+cluster.
+
+The environment needs to be adjusted to allow the tests to operate against
+such a cluster.
+The following high level steps are required to accomplish this:
+- Clone kubevirt repository on the machine that has access to the cluster.
+- Identify the deployed kubevirt version.
+- From the git repo, checkout to the commit that correlates to the running
+kubevirt cluster.
+- Run the tests by executing the
+  [./tests/run-tests-on-external-cluster.sh](./run-tests-on-external-cluster.sh)
+  script.
+
+> **Note:** In case the external cluster uses dynamic storage provisioner
+such as hostpath, then there is a need to provide a specific custom storage
+configuration through a configuration file.
+This is currently not covered by this document and the script helper.
+
+### Identify the commit on which the deployed kubevirt cluster is based on
 
-There is a make target to run this with the config
-taken from hack/config.sh:
+In order to identify on which commit the current kubevirt cluster is based on,
+there is a need to query a kubevirt infra pod and then inspect its base
+container.
 
+- Lookup for the `virt-operator` pod:
+  `kubectl get pods --all-namespaces | grep "virt-operator"`
+- Use the full pod name and its namespace to inspect the container it uses:
+  `kubectl describe <virt-operator full name> -n <virt-operator namespace>`
+- Identify and record the pod image:
+  ```
+  Containers:
+  virt-operator:
+    Image:         <reference to image on the registry>
+  ```
+- Inspect the image to retrieve the commit:
+  `skopeo inspect docker://<image path>`
+  The commit ID is set on the `upstream-vcs-ref`.
+- Use git-checkout to sync the sources to the relevant commit ID.
+  `git checkout <commit id>`
+
+At this stage, the tests sources should be in sync with the running cluster
+and it is safe to setup the configuration and run the tests.
+
+### Run the tests
+
+The test environment needs to be configured to work with the running cluster
+and the test can then be build and executed against it.
+
+These steps have been automated by the
+[run-tests-on-external-cluster.sh](./run-tests-on-external-cluster.sh) script.
+
+Note that the namespace in which kubevirt components run on the cluster
+needs to be specified.
```

Comment:
We should also list the dependencies on the cluster. We expect CDI and KubeVirt to be deployed.

Reply:
Line 64 specify kubevirt, that is almost trivial.
CDI we do not expect nor deploy AFAIK, see Roman and Igor conversation.